modules/drivers/sparksql/src/metabase/driver/sparksql.clj

```diff
@@ -88,12 +88,19 @@
   (when s
     (s/replace s #"-" "_")))
 
+(defn- valid-describe-table-row [row]
+  (not(or
+       (s/blank? (:col_name row))
+       (s/starts-with? (:col_name row) "#") ;; Hive describe table result has commented rows to distinguish partitions
+       (s/blank? (:data_type row))
+       (s/starts-with? (:data_type row) "#"))))
+
 ;; workaround for SPARK-9686 Spark Thrift server doesn't return correct JDBC metadata
 (defmethod driver/describe-database :sparksql [_ {:keys [details] :as database}]
   {:tables (with-open [conn (jdbc/get-connection (sql-jdbc.conn/db->pooled-connection-spec database))]
              (set (for [result (jdbc/query {:connection conn}
                                            ["show tables"])]
-                    {:name   (:tablename result)
+                    {:name   (if (contains? result :tablename) (:tablename result) (:tab_name result)) ;; column name differ depending on server (SparkSQL, hive, Impala)
```

Comment:
I would use (`some`)[https://clojuredocs.org/clojure.core/some] here

```clj
(some result [:tablename :tab_name])
```

to get either `:tablename` or `:tab_name` (whichever is truthy, if any)